### PR TITLE
Add node for the added field

### PR DESCRIPTION
**Pull request by iFlameing**
Related to #180 

I will refactor all the code and squash all my commit into one once I am done with this feature. I just want to complete the POC of this feature.

### CODE REVIEW

**Comment by datakurre** [comment]
output

File: src/gatsby-node.js [hunk 1]
```diff
@@ -371,21 +373,38 @@ exports.sourceNodes = async (
...
-          socket.emit('message',{data:'I am Excited'})
-
-        },5000);
-        socket.on('setTimeout',(data) => {
-          console.log("this is from the json payload after some timeout");
-          console.log(data)
-        })
-    })
+  if(websocketUpdates) {
+    let ws = new WebSocket('ws://localhost:8080/Plone/');
```
Comment (datakurre) [inline src/gatsby-node.js]: maybe baseUrl with protocol changed from http/https to ws?

**Comment by datakurre** [comment]
By just applying logic from node created to modified, this seems to work for updating the existing pages. Challenge is to update all nodes (changed, parent and also affected navigation and breadcrumb nodes).

I'm still not sure what should be done to trigger page builds. gatsby-starter-blog seems to do nothing special, but it just works

**Comment by datakurre** [comment]
@iFlameing I rechecked and your code already works. New node and also new page is created. I just that because the navigation nodes are not updates, you don't see the update.

**Comment by datakurre** [comment]
In addition, there's issue that Plone is currently sending all updates, not applying security checks, causing the plugin to receive updates also for private pages.

I'll try to fix that next week (it is not completely trivial), but also you should eventually write your code so that gatsby-source-plone would not crash on receiving HTTP 401 from Plone.

**Comment by iFlameing** [comment]
> I just that because the navigation nodes are not updates, you don't see the update.

I didn't understand this statement.

 I am unable to see the new page when I am adding a new node. can you share the steps which you did to see the new page?

I am understanding the `ploneNodeGenerator` code because Currently, I am only making contentnode using `makecontentnode` function. Once I understand the ploneNodeGenerator code I can simply call it with the given url and we can update all node (changed, parent and also affected navigation and breadcrumb nodes).

**Comment by datakurre** [comment]
After creating node with your current code, you should see the new node in GraphQL-console and the new page in 404 page (just type nonexisting path into browser).

Rendering page for the new page fails, though, because there is no matching breadcrumbs node yet.

Take your time in learning how those functions using async iterators work. This already works better than I expected so you should have plenty of time for learning and refactoring.

**Comment by iFlameing** [comment]
@datakurre can you review this once again? I successfully tested the creation of breadcrumbs and navigation node and updating the parents and all related child node :)

**Comment by datakurre** [comment]
Wow, that sounds great! I’ll check this with time on Tuesday evening.

**Comment by datakurre** [comment]
@iFlameing I see you learned to use the async generators there. Great!

I also confirm that this works for me. Let's merge this, so you could continue for initial implementations for also modified and deleted.

**Comment by datakurre** [comment]
@iFlameing One more for later pull request. Please, run "make format" before commit. I will probably add that to Travis CI so that Travis-CI will be read, if "make format" would do changes.

**Pull request merged**